### PR TITLE
[Kernel][Data skipping] Add support data skipping for IS_NULL and NOT(IS_NULL)

### LINKED ISSUE #2530: [Kernel][Data skipping] Add support data skipping for IS_NULL and NOT(IS_NULL)
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [X] Kernel
- [ ] Other (fill in here)

### Overview

Currently Kernel supports a limited set of expressions. We should use file statistics to prune files based on the `IS_NULL` expression.

### Motivation

Better file pruning.

### Further details

This means we should
1) add IS_NULL to the Kernel [Predicate](https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java) and support it in the kernel-defaults project (completed as part of #2651).
2) Generate a data skipping filter according to the [same rules](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/stats/DataSkippingReader.scala#L467) we use in delta-spark

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [x] No. I cannot contribute this feature at this time.

### PR DESCRIPTION
**Pull request by tlm365**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

## Description
Resolves #2530

## How was this patch tested?
Unit tests added.

## Does this PR introduce _any_ user-facing changes?
No.

### CODE REVIEW

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java [hunk 1]
```diff
@@ -241,6 +241,31 @@ private static Optional<DataSkippingPredicate> constructDataSkippingFilter(
...
+                        Column nullCountCol = schemaHelper.getNullCountColumn(childColumn);
+                        Literal zero = Literal.ofLong(0);
+                        return Optional.of(
+                            new DataSkippingPredicate(
+                                ">",
+                                Arrays.asList(nullCountCol, zero),
+                                new HashSet<Column>(){
+                                    {
+                                        add(nullCountCol);
+                                    }
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java:261]:
```suggestion
                                Collections.singleton(nullCountCol)
```

**Review by vkorukanti — COMMENTED** [review]
@allisonport-db could you take a look?

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala [hunk 2]
```diff
@@ -1304,15 +1304,15 @@ class ScanSuite extends AnyFunSuite with TestUtils with ExpressionTestUtils with
...
+          not(isNotNull(col("struct_col"))) // we don't skip on non-leaf columns
         ),
-        misses = Seq(equals(col("id"), ofInt(1)))
+        misses = Seq(
+          equals(col("id"), ofInt(1)),
+          isNull(col("id")),
+          isNull(col("arr_col")),
+          isNull(col("map_col")),
+          isNull(nestedCol("struct_col.field1"))
+        )
```
[the remark this answers is not in the document]
Comment (allisonport-db) [reply kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala:1315]: You can include those above plus the ones you added

**Review by allisonport-db — COMMENTED** [review]
Can you add an additional

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlm365) [reply kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala:1315]: @allisonport-db TYSM

**Review by allisonport-db — APPROVED** [review]
LGTM thanks for doing this!

**Pull request merged**